console/src/main.rs

```diff
@@ -15,6 +15,25 @@ mod tasks;
 mod term;
 mod view;
 
+enum ConnectionState {
+    Connected,
+    Disconnected,
+}
+
+async fn connect(
+    target: String,
+    should_delay: bool,
+) -> Result<tonic::Streaming<TaskUpdate>, Box<dyn std::error::Error>> {
+    if should_delay {
+        tokio::time::sleep(std::time::Duration::from_millis(1000)).await;
+    }
+
+    let mut client = TasksClient::connect(target).await?;
+    let request = tonic::Request::new(TasksRequest {});
+    let rsp = client.watch_tasks(request).await?;
+    Ok(rsp.into_inner())
+}
```

Comment:
eventually i imagine we might want to move the connection-management code out of `main.rs` into its own module? since there's not a ton of it right now, it's fine here for now, but just a thought.

Reply:
Yeah I think so, and if we do that, it seems to make sense to beef it a bit to provide things like backoff